frontend/pages/views/MapPageView.js

```diff
@@ -0,0 +1,285 @@
+import * as PropTypes from "prop-types";
+
+function MapPageEntryLogo(){
+    return (<div>
+            <Container id="map-page-title">
+                <Row>
+                    <Col>
+                        <Logo id="site-logo" top={22} left={-1.5} logo_type={"title-logo"}/>
+                        <h2 className="page-title">Map</h2>
+                    </Col>
+                </Row>
+            </Container>
+            </div>
+    );
+}
+
+function MapSquareList(props){
+    //Given an arrondissiment, only keeps the map squares that actually contain photos
+    const mapSquareNumbers = props.arrondissementData[props.arrondissementNumber-1]["map_square_numbers"].filter(
+        number => props.filledMapSquares.has(number));
+
+    if(mapSquareNumbers.length === 0)
+        return <span>No Map Squares!</span>;
+
+    //Returns a comma delimited list of the map square numbers in a arrondissement. Each number
+    //links to corresponding map page
+    const displayedMapSquares = mapSquareNumbers.map((map_square_number, i) => {
+        const link = "/map_square/"+map_square_number+"/";
+        return (<span key={map_square_number}>
+                        <a href={link} className="info-list-link">{map_square_number}</a>
+                        {mapSquareNumbers[i+1] ? ", " : ""}
+                </span>);
+    });
+
+    return(<span>{displayedMapSquares}</span>);
+}
+
+function densityOverlay(mapData) {
+    const sortedMapData = Object.values(mapData)
+    .sort((a, b) => a.num_photos - b.num_photos);
+    // Gets the max number of photos in a single square out of all squares to form buckets later
+    const maxNumOfPhotos = sortedMapData && sortedMapData.length
+        ? sortedMapData[sortedMapData.length - 1].num_photos
+        : 0;
+    // Creating 5 buckets based on lowest to highest number of photos per square
+    const twentyPctMax = Math.round(0.2 * maxNumOfPhotos);
+    const fortyPctMax = Math.round(0.4 * maxNumOfPhotos);
+    const sixtyPctMax = Math.round(0.6 * maxNumOfPhotos);
+    const eightyPctMax = Math.round(0.8 * maxNumOfPhotos);
+    const buckets = [0, twentyPctMax, twentyPctMax + 1,
+        fortyPctMax, fortyPctMax + 1,
+        sixtyPctMax, sixtyPctMax + 1,
+        eightyPctMax, eightyPctMax + 1, maxNumOfPhotos];
+    const layer = (<>
+        <Legend buckets={buckets}/>
+
+        {
+            sortedMapData.map((mapSquareData) => {
+                const index = mapSquareData.number;
+                const coords = mapSquareData.topLeftCoords;
+                const numberOfPhotos = mapSquareData.num_photos;
+
+                const mapSquareBounds = [
+                    [coords.lat, coords.lng],
+                    [coords.lat - MAPSQUARE_HEIGHT, coords.lng - MAPSQUARE_WIDTH]
+                ];
+                const link = "/map_square/" + index;
+                let mapSquareBucket = "";
+                // set of conditionals to calculate photo density for heat map
```

Comment:
Just FYI: If we were to keep this, this would have to be refactored out into its own component/set of functions. Otherwise we're just repeating ourselves. 
However, @ADJohnson314 is currently working on making this calculation on the backend so that we only have to calculate this once as opposed to every time the map loads.